pkg/util/priority/priority_test.go

```diff
@@ -0,0 +1,171 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package priority
+
+import (
+	"context"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/constants"
+	"sigs.k8s.io/kueue/pkg/util/pointer"
+
+	"testing"
+)
+
+func TestPriority(t *testing.T) {
+	tests := []struct {
+		desc     string
+		workload *kueue.Workload
+		expected int32
+	}{
+		{
+			desc: "priority is specified",
+			workload: &kueue.Workload{
+				Spec: kueue.WorkloadSpec{
+					Priority: pointer.Int32(100),
+				},
+			},
+			expected: 100,
+		},
+		{
+			desc: "priority is empty",
+			workload: &kueue.Workload{
+				Spec: kueue.WorkloadSpec{},
+			},
+			expected: constants.DefaultPriority,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			got := Priority(tt.workload)
+			if got != tt.expected {
+				t.Errorf("Priority does not match: got: %d, expected: %d", got, tt.expected)
+			}
+		})
+	}
+}
+
+func TestGetPriorityFromPriorityClass(t *testing.T) {
+	scheme := runtime.NewScheme()
+	if err := schedulingv1.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding scheduling scheme: %v", err)
+	}
+
+	tests := []struct {
+		desc              string
+		priorityClassList *schedulingv1.PriorityClassList
+		priorityClassName string
+		expectedName      string
+		expectedValue     int32
+		expectedErr       bool
+	}{
+		{
+			desc: "priorityClass is specified and it exists",
+			priorityClassList: &schedulingv1.PriorityClassList{
+				Items: []schedulingv1.PriorityClass{
+					{
+						ObjectMeta: v1.ObjectMeta{Name: "test"},
+						Value:      50,
+					},
+				},
+			},
+			priorityClassName: "test",
+			expectedName:      "test",
+			expectedValue:     50,
+		},
+		{
+			desc: "priorityClass is specified and it does not exist",
+			priorityClassList: &schedulingv1.PriorityClassList{
+				Items: []schedulingv1.PriorityClass{},
+			},
+			priorityClassName: "test",
+			expectedErr:       true,
+		},
+		{
+			desc: "priorityClass is unspecified and one global default exists",
+			priorityClassList: &schedulingv1.PriorityClassList{
+				Items: []schedulingv1.PriorityClass{
+					{
+						ObjectMeta:    v1.ObjectMeta{Name: "globalDefault"},
+						GlobalDefault: true,
+						Value:         40,
+					},
+				},
+			},
+			expectedName:  "globalDefault",
+			expectedValue: 40,
+		},
+		{
+			desc: "priorityClass is unspecified and multiple global default exist",
```

Comment:
Is it possible that priorityClass is unspecified and no global default exists?

Reply:
Maybe. Then it will use `constants.DefaultPriority`